### DISCUSSION TITLE
Add a `strict` option to `presetMini`

### DISCUSSION THREAD

**Discussion by zyyv** [unocss/unocss#1220] [Ideas]
UnoCSS engine extracts property by `extractorSplit`
https://github.com/unocss/unocss/blob/0709b91a2f35bd04ef132bb911d400a2398ad426/packages/core/src/extractors/split.ts#L4-L12

The regex is too forcely for `presetMini`.
Like this case: [Playground](https://uno.antfu.me/play/?html=DwEwlgbgBALgpgZxgXgEQHsAOBaArABlQD4AoKKAVwDt0BjBBE4AenAiKA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCqWEcAvnOlBCHAOQCujAYypVuAKFGoAHpFgoMAQ14AbeGkw58hYEQAUCUXDhRl1AFxwA2he4DeVGJwC0xpam5kkAiEujnulZG4WAF1gkgM4KgALaBhbGCpzfUNDGzsHEEdo2PjuPxgpGEclIjgCySLoeSwiVDgYgDdUKFNyooL5JXcI5nDDCmpaRMsBmnpGHQBKMNFmSdEgA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g). It will produce unnecessary css.

I think we should only parse html and extract property in `class` attribute. 

How about add a `strict` option in `presetMini`, like `presetAttributify`.

**Comment by antfu** [comment]
How to do we handle this with `strict`?

```html
<script setup>
const btnStyle = 'bg-blue px1 py2'
</script>

<template>
  <div :class="btnStyle"/>
</template>
```

**Reply by zyyv** [reply]
Oh, I didn't think about it. 😅
